metrics/resource_view.go

```diff
@@ -57,8 +58,22 @@ type meters struct {
 // optionForResource, which must lock allMeters, but only needs to lock
 // resourceViews if a new meter needs to be created.
 var resourceViews = storedViews{}
-var allMeters = meters{
-	meters: map[string]*meterExporter{"": &defaultMeter},
+var allMeters = meters{}
+
+// Sizing is largely arbitrary - selected to easily cover a realistic number of revisions
+const cacheSize int = 4096
+
+func init() {
+	// lru.New() only throws on negative size
+	// The default meter is required - make sure it's not evicted
+	var cache *lru.Cache
+	cache, _ = lru.NewWithEvict(cacheSize, func(key interface{}, value interface{}) {
+		if key == "" {
+			cache.Add(key, value)
```

Comment:
Codecov complains that this branch/code is not tested. I'd normally say "meh", but in this case, it seems fairly critical.

Reply:
This section (this section's logical replacement, really) has coverage, now.